### PR TITLE
Quic proxy gm improvements

### PR DESCRIPTION
**Pull request by grooviegermanikus**
Changes based on feedback here https://github.com/blockworks-foundation/lite-rpc/pull/164#discussion_r1313079727

### CODE REVIEW

File: quic-forward-proxy/src/inbound/proxy_listener.rs [hunk 1]
```diff
@@ -49,17 +44,9 @@ impl ProxyListener {
...
-                match Self::accept_client_connection(
-                    connection,
-                    forwarder_channel_copy,
-                    exit_signal,
-                )
-                .await
-                {
+                match Self::accept_client_connection(connecting, forwarder_channel_copy).await {
                     Ok(()) => {
                         debug!("connection handles correctly");
```
Comment (godmodegalactus) [inline quic-forward-proxy/src/inbound/proxy_listener.rs:51]: Is is normal that after accepting connection we do not do anything?

File: quic-forward-proxy/src/outbound/tx_forward.rs [hunk 2]
```diff
@@ -218,16 +216,23 @@ pub async fn tx_forwarder(
...
     let mut to_shutdown = Vec::new();
     for (tpu_address, handle) in &*agents {
         if tpu_address == current_tpu_address {
             continue;
         }
 
-        let last_used_ms = handle.age_ms.load(Ordering::Relaxed);
+        let unused_period = {
+            let last_used_at = handle.last_used_at.read().await;
+            now - *last_used_at
```
Comment (godmodegalactus) [inline quic-forward-proxy/src/outbound/tx_forward.rs:232]: should be saturating_sub instead/

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

File: quic-forward-proxy/src/inbound/proxy_listener.rs (hunk 1 above)
Comment (grooviegermanikus) [reply quic-forward-proxy/src/inbound/proxy_listener.rs:51]: code is IMO okey, but naming is bad - will change _accept_ to _handle_

File: quic-forward-proxy/src/outbound/tx_forward.rs (hunk 2 above)
Comment (grooviegermanikus) [reply quic-forward-proxy/src/outbound/tx_forward.rs:232]: nono . that's Instant - Instant

Comment (godmodegalactus) [reply quic-forward-proxy/src/outbound/tx_forward.rs:232]: Ahh yes.

File: quic-forward-proxy/src/util.rs [hunk 3]
```diff
@@ -15,27 +13,3 @@ where
 {
     tokio::time::timeout(FALLBACK_TIMEOUT, future)
 }
-
```
Comment (godmodegalactus) [inline quic-forward-proxy/src/util.rs:18]: this file should be removed right?

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

File: quic-forward-proxy/src/util.rs (hunk 3 above)
Comment (grooviegermanikus) [reply quic-forward-proxy/src/util.rs:18]: why that? should I inline the timeout?

Comment (godmodegalactus) [reply quic-forward-proxy/src/util.rs:18]: Ok i guess its fine.

**Review by godmodegalactus — APPROVED** [review]

**Pull request merged**